### ISSUE TITLE
Tasks for Spring Boot 2.7.1

### ISSUE THREAD

**Issue by Netyyyy** [Azure/azure-sdk-for-java#29658]
### Context

Spring Boot released [2.7.1](https://github.com/spring-projects/spring-boot/releases/tag/v2.7.1).

We should upgrade external dependencies' version align with [spring-boot-dependencies:2.7.1](https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/2.7.1/spring-boot-dependencies-2.7.1.pom) managed versions. 

### Goal

1. Make our libraries compatible with latest Spring Boot: 2.7.1 by making external dependencies' version align to spring-boot-dependencies:2.7.1 managed versions

### Solution

Tasks are copied from this page: https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Cloud-Azure-tasks-for-a-new-Spring-Boot-release

- Before azure-sdk-for-java in band release
  - [x] Create a new issue to track the following tasks. Sample: [#28498](https://github.com/Azure/azure-sdk-for-java/issues/28498). And add the new created issue into [#28518](https://github.com/Azure/azure-sdk-for-java/issues/28518).
  - [x] Read spring boot release notes. Sample: [Spring-Boot-2.6-Release-Notes](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.6-Release-Notes). Write down items we(Spring Cloud Azure team members) should care about. Just write these items by adding a new comment in the GitHub issue created in previous step.
  - [x] Upgrade external dependencies' version according to spring-boot-dependencies and spring-cloud-dependencies. Refer to [related readme](https://github.com/Azure/azure-sdk-for-java/blob/aec4c6247ba7ba4de57dd866e3f5511ca4fbd387/sdk/spring/scripts/README.md). **NOTE**: This step should be finished before `Core code complete`. You can get more information about `Core code complete` in the Email with subject like `APRIL Release Kickoff`. https://github.com/Azure/azure-sdk-for-java/pull/29638

- After azure-sdk-for-java in band release.

  Update spring-boot and spring-cloud version in these places:

  - [x] [azure-spring-boot-samples](https://github.com/Azure-Samples/azure-spring-boot-samples)https://github.com/Azure-Samples/azure-spring-boot-samples/pull/410
  - [x] [Reference doc](https://github.com/microsoft/spring-cloud-azure/tree/docs/docs/src/main/asciidoc).
  - [x] [MS docs](https://docs.microsoft.com/en-us/azure/developer/java/spring-framework/). https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/3255
  - [x] [Wiki page: Spring Cloud Azure Timeline](https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Cloud-Azure-Timeline)
  - [x] [Wiki page: Spring Versions Mapping](https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Versions-Mapping)
  - [ ] ~~[Wiki page: Spring Support Multiple Versions](https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Support-Multiple-Versions)~~
